packages/google-compute-engine-oslogin/google_oslogin_control

```diff
@@ -314,16 +320,24 @@ restart_sshd() {
     return 0
   fi
   echo "Restarting SSHD"
-  for svc in "ssh" "sshd"; do
-    restart_service "$svc"
-  done
+  if is_freebsd; then
```

Comment:
'ssh' doesnt exist most of the time, either, we just try it in case. does it have a negative effect in freebsd?

Reply:
Yes, it does. If we try to restart the `ssh` service the script returns non zero, which will not allow OSLogin activation.

Reply:
what script returns non zero? we try systemd, `service` and then init scripts. but in each case we check for existence and status first so that non-existent services should have no impact. if you can share the specific details i could probably make an improvement here

Reply:
Sorry for the delay, I forgot to reply this message. The problem here is that the `restart_service` function does not handle well FreeBSD services, explanation below:

```
if ! service "$service" status 2>&1 | grep -q unrecognized; then
    service "$service" restart
    return $?
else
    return 0
fi
```
The snippet above try to find the `unrecognized` string in the output but this string does not exist in FreeBSD output when trying to check the status of a non existent service:

```
kanashiro@kanashiro-freebsd-13-debug:~ $ sudo service ssh status
ssh does not exist in /etc/rc.d or the local startup
directories (/usr/local/etc/rc.d), or is not executable
kanashiro@kanashiro-freebsd-13-debug:~ $ echo $?
1
kanashiro@kanashiro-freebsd-13-debug:~ $ sudo service sshd status
sshd is running as pid 31038.
kanashiro@kanashiro-freebsd-13-debug:~ $ echo $?
0
```
The best approach would be to improve the 'restart_service` function indeed, any idea about the best way to do it? Check also for `does not exist` string?

Reply:
sorry, I was using the FreeBSD ports package which is synced with `master` and does not contain the modification I proposed above (it's already implemented in `development`). So, we need to wait for the next release to get it into FreeBSD package (now it is fixed via patch).

Reply:
the simplest solution is to also check for 'does not exist', yeah. checking BSD [manpage][1] i see there are `-e` and `-r` arguments for explicit checking, we could check that list first. 

[1]: https://www.freebsd.org/cgi/man.cgi?query=service&manpath=FreeBSD+12.0-RELEASE+and+Ports